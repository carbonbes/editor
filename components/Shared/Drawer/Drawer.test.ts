import { describe, expect, test } from 'vitest'
import { flushPromises, mount } from '@vue/test-utils'
import { DrawerOverlay, DrawerPortal, DrawerRoot } from './'

const Component = defineComponent({
  components: {
    DrawerOverlay,
    DrawerPortal,
    DrawerRoot,
  },
  template: `
    <DrawerRoot>
      <DrawerPortal>
        <DrawerOverlay />
      </DrawerPortal>
    </DrawerRoot>
  `,
})

describe('Drawer', () => {
  test('should render the DrawerOverlay', async () => {
    const wrapper = mount(Component, { props: { open: true } })
    await flushPromises()

    const overlay = wrapper.getComponent(DrawerOverlay)

    expect(overlay.html()).toMatchSnapshot()

    wrapper.unmount()
  })
})
