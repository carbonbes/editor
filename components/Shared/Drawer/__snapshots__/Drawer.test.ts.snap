// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Drawer > should render the DrawerOverlay 1`] = `"<div data-state="open" style="pointer-events: auto;" vaul-drawer-visible="false" vaul-overlay="" vaul-snap-points="false" vaul-snap-points-overlay="true" class="fixed inset-0 bg-black/25 backdrop-blur-[2px]"></div>"`;
